torch/_dynamo/backends/registry.py

```diff
@@ -94,22 +96,27 @@ def _lazy_import():
 
     assert dynamo_minifier_backend is not None
 
+    _register_entrypoint_backends()
+
 
 @functools.lru_cache(None)
-def _lazy_import_entry_point(backend_name: str):
+def _register_entrypoint_backends():
     from importlib.metadata import entry_points
 
-    compiler_fn = None
     group_name = "torch_dynamo_backends"
     if sys.version_info < (3, 10):
-        backend_eps = entry_points()
-        eps = [ep for ep in backend_eps.get(group_name, ()) if ep.name == backend_name]
-        if len(eps) > 0:
-            compiler_fn = eps[0].load()
+        eps = entry_points()
+        eps = eps[group_name] if group_name in eps else []
+        eps = {ep.name: ep for ep in eps}
     else:
-        backend_eps = entry_points(group=group_name)
-        if backend_name in backend_eps.names:
-            compiler_fn = backend_eps[backend_name].load()
-
-    if compiler_fn is not None and backend_name not in list_backends(()):
-        register_backend(compiler_fn=compiler_fn, name=backend_name)
+        eps = entry_points(group=group_name)
+        eps = {name: eps[name] for name in eps.names}
+    for backend_name in eps:
+        try:
+            compiler_fn = eps[backend_name].load()
```

Comment:
I am not convinced this is a good idea.  My concern is that it is importing many third party modules not requested by the user, and those third part modules could either have side effects at import time or have bugs in them.

There are possibly security concerns with this, if the user manages to install a malicious package now calling into `torch.*` will execute that unrelated malicious package.  While before this change you would need to pass the name of the that package to pytorch.

There are also versioning concerns, suppose the user has some older (or newer) backend installed that is incompatible with the current pytorch version.  Now pytorch will try to import that which could lead to errors at import time.

Importing some packages also can have side effects that might reconfigure pytorch in a way the user doesn't expect.  This will lead to super hard to diagnose bugs.

I'd suggest instead either:
1) Make this behavior more clear in the documentation
2) Modify list_backends() to get the name of backend packages *without* importing them until they are needed.

Reply:
that makes sense -- I'll get the backend names eagerly without actually calling `eps[backend_name].load()` until later. I'll try to get this updated tomorrow, but if not I'll be on PTO for a bit so might need to come back to this later.

Reply:
updated to track the backend names + optional EntryPoint and compiler_fn's separately